invenio_app_ils/ui/src/invenio_app_ils/common/components/SearchBar/SearchBar.js

```diff
@@ -1,35 +1,59 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { Input } from 'semantic-ui-react';
+import { QueryBuildHelper } from './components/QueryBuildHelper/';
 
 export class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  onChangeHandler = (e, { value }, onInputChange) => {
+    onInputChange(value);
+  };
+
+  onKeyPressHandler = (event, executeSearch) => {
+    if (event.key === 'Enter') {
+      executeSearch();
+    }
+  };
+
   render() {
     const {
       currentQueryString,
       onInputChange,
       executeSearch,
       placeholder,
+      queryHelperFields,
     } = this.props;
     return (
-      <Input
-        action={{
-          content: 'Search',
-          onClick: () => {
-            executeSearch();
-          },
-        }}
-        fluid
-        placeholder={placeholder}
-        onChange={(e, { value }) => {
-          onInputChange(value);
-        }}
-        value={currentQueryString}
-        onKeyPress={event => {
-          if (event.key === 'Enter') {
-            executeSearch();
+      <>
+        <Input
+          action={{
+            content: 'Search',
+            onClick: () => {
+              executeSearch();
+            },
+          }}
+          fluid
+          placeholder={placeholder}
+          onChange={(e, { value }) =>
+            this.onChangeHandler(e, { value }, onInputChange)
```

Comment:
I think here is ok to be like that @topless as it is just for the event handler to be happy and to be able to pass extra params when the event is triggered. The actual handler is already another function.

Reply:
yes, I left this one like that for this exact reason